linalg/ode.hpp

```diff
@@ -931,6 +938,100 @@ class WBZAlphaSolver : public GeneralizedAlpha2Solver
 
 };
 
+/// Class for solving systems of split ODEs:
+/// M dx/dt = f_1(x,t) + f_2(x,t)
+/// where M^{-1}*f_1 and M^{-1}*f_2 are treated differently (e.g.,
+/// explicitly and implicitly).This is the abstract base class for
+/// such ODE solvers expecting a TimeDependentOperator with split
+/// functionality. Setting EvalMode=ADDITIVE_TERM_1 and calling
+/// Mult(...) should return k1=M^{-1}*f_1(x,t). Setting
+/// EvalMode=ADDITIVE_TERM_2 and calling ImplicitSolve(...) should
+/// solve M*k2 = f_2(x+gamma*k2,t).
+class SplitODESolver : public ODESolver
```

Comment:
```suggestion
class ARKODESolver : public ODESolver // Additive Runge Kutta method
private:
const real_t *a_imp, *b_imp, *c_imp; // implicit Butcher tableau coefficients
const real_t *a_exp, *b_exp, *c_exp; // explicit Butcher tableau coefficients
```
The ARS(2,2,2) (in `class IMEXRK2`), ARS(2,3,2) (in `class IMEXRK2_3StageExplicit`) and ARS(3,4,3) (in `class IMEX_DIRK_RK3`) can be implemented/unified as a general ARK IMEX scheme (Eqs. 2.1b - 2.1e in [this paper](https://steveruuth.org/wp-content/uploads/2020/10/ars.pdf)) by using specific explicit and implicit Butcher tableaus. See [ExplicitRKSolver](https://docs.mfem.org/html/classmfem_1_1ExplicitRKSolver.html#:~:text=mfem-,ExplicitRKSolver,-Public%20Member%20Functions) to see how a tableau is used. It's better to implement the general ARK using tableaus so users can use different ARK IMEX schemes by supplying different tableaus.

Reply:
A general class like that is implemented in #2474 , so it could be just reused perhaps.